### PR TITLE
Add support for src/common directory

### LINKED ISSUE #1037: make a "src/common/" directory (package?) to share between server and acquisition code
to promote modularity and reduce code repetition, we should create a "`./src/common/`" directory (and potentially package too).  this will require some changes to our docker definitions and maybe also other parts of our build process.

the first file/module to include here could be our structured logger code...  see: https://github.com/cmu-delphi/delphi-epidata/pull/1027#discussion_r1022131246_

### PR DESCRIPTION
**Pull request by rzats**
Closes #1037.

**Prerequisites**:

- [x] Unless it is a documentation hotfix it should be merged against the `dev` branch
- [x] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [x] Build is successful
- [x] Code is cleaned up and formatted

### Summary

Adds support for the `src/common` directory visible by both `acquisition` & `server` and restructures the web Docker container's imports to make them make a little more sense. This includes modifications to the:
- `ci.yaml` workflow
- `deploy.json`
- `Makefile`
- `Dockerfile`

Additionally, this merges the two (identical) logger files from `acquisition` and `server` into one, now located in the `common` directory.

### CODE REVIEW

**Comment by rzats** [comment]
@melange396 @krivard this is potentially an *extremely* breaking change, so if there's any way to test this besides making sure that the usual tests pass (which they do), please let me know!

**Comment by melange396** [comment]
we can try deploying to staging, which should be a much closer approximation to the production environment than the dev docker images are, right @korlaxxalrok ?

File: devops/Dockerfile [hunk 1]
```diff
@@ -8,14 +8,16 @@ LABEL org.opencontainers.image.source=https://github.com/cmu-delphi/delphi-epida
...
+COPY ./src/server /app/delphi/epidata/server
+COPY ./src/common /app/delphi/epidata/common
+COPY --from=builder ./src/build/lib/ /app/delphi/epidata/lib/
 
 COPY requirements.api.txt /app/requirements_also.txt
 
 RUN ln -s -f /usr/share/zoneinfo/America/New_York /etc/localtime \
-      && rm -rf /app/app/__pycache__ /app/app/*.php \
-      && chmod -R o+r /app/app \
+      && rm -rf /app/delphi/epidata/__pycache__ /app/delphi/epidata/*.php \
```
Comment (krivard) [inline devops/Dockerfile:19]: question: do we still need this line?

(i thought #1092 got rid of all the php files)

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -61,7 +61,7 @@ jobs:
         run: |
           docker network create --driver bridge delphi-net
           docker run --rm -d -p 13306:3306 --network delphi-net --name delphi_database_epidata --cap-add=sys_nice delphi_database_epidata
-          docker run --rm -d -p 10080:80 --env "SQLALCHEMY_DATABASE_URI=mysql+mysqldb://user:pass@delphi_database_epidata:3306/epidata" --env "FLASK_SECRET=abc" --env "FLASK_PREFIX=/epidata" --network delphi-net --name delphi_web_epidata delphi_web_epidata
+          docker run --rm -d -p 10080:80 --env "MODULE_NAME=delphi.epidata.server.main" --env "SQLALCHEMY_DATABASE_URI=mysql+mysqldb://user:pass@delphi_database_epidata:3306/epidata" --env "FLASK_SECRET=abc" --env "FLASK_PREFIX=/epidata" --network delphi-net --name delphi_web_epidata delphi_web_epidata
```
Comment (krivard) [inline .github/workflows/ci.yaml:64]: question: where is `MODULE_NAME` used?

Comment (melange396) [reply .github/workflows/ci.yaml:64]: its used in the base image that gets built upon in `devops/Dockerfile`, "`tiangolo/meinheld-gunicorn`".  see [the documentation](https://github.com/tiangolo/meinheld-gunicorn-docker#module_name) for that image, which i also [mentioned](https://github.com/cmu-delphi/delphi-epidata/issues/1037#issuecomment-1416172115) in the associated issue (#1037) for this pr.

comments noting the purpose of `MODULE_NAME` along with that documentation link above should probably be included here, in the `Makefile`, and in `devops/Dockerfile`.

`MODULE_NAME` will also need to be populated in automation or ansible or whatever it is that runs the API server code, which we will probably see when we try to test running this on our staging and/or qa setups. (cc: @korlaxxalrok )

**Review by melange396 — CHANGES_REQUESTED** [review]
can we move `CovidcastRow` into `common` as well?  that could help with #1103, as referenced in #1078.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (korlaxxalrok) [reply .github/workflows/ci.yaml:64]: Yep, looks like we'd need to pass an additional environment variable.

**Comment by korlaxxalrok** [comment]
This looks good to me. Since this touches both acquisition and the API container image we should coordinate and test fully on staging.

**Review by korlaxxalrok — APPROVED** [review]
Looks good to me (pending suggestions + any remaining questions) 👍

**Comment by rzats** [comment]
@korlaxxalrok @melange396 @krivard applied the changes, re-requesting review!

(Also, @korlaxxalrok, let me know how and when you'd like to test this on staging)

File: tests/acquisition/covidcast/test_covidcast_row.py [hunk 3]
```diff
@@ -14,7 +14,7 @@
 )
 
 # py3tester coverage target (equivalent to `import *`)
-__test_target__ = 'delphi.epidata.acquisition.covidcast.covidcast_row'
+__test_target__ = 'delphi.epidata.common.covidcast_row'
```
Comment (melange396) [inline tests/acquisition/covidcast/test_covidcast_row.py:17]: can we move this file to `tests/common/test_covidcast_row.py` to match the `src/common/` structure?

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -57,11 +57,12 @@ jobs:
           docker build -t delphi_web_epidata -f ./devops/Dockerfile .
           cd ../../../
 
+      # MODULE_NAME is used to set the location of the main.py file, see https://github.com/tiangolo/meinheld-gunicorn-docker#module_name
```
Comment (melange396) [inline .github/workflows/ci.yaml:60]: how about this text?  making note of the variable name is important:
```suggestion
      # MODULE_NAME specifies the location of the `app` variable, the actual WSGI application object to run.
      # see https://github.com/tiangolo/meinheld-gunicorn-docker#module_name
```

**Review by melange396 — CHANGES_REQUESTED** [review]

**Review by melange396 — APPROVED** [review]
wonderful!

**Review by korlaxxalrok — APPROVED** [review]
Yep, looks. We can merge and test on staging next.

**Pull request merged**